feature/gnoi/os/tests/osinstall/osinstall_test.go

```diff
@@ -257,29 +224,49 @@ func (tc *testCase) transferOS(ctx context.Context, t *testing.T, standby bool)
 // verifyInstall validates the OS.Verify RPC returns no failures and version numbers match the
 // newly requested software version.
 func (tc *testCase) verifyInstall(ctx context.Context, t *testing.T) {
-	r, err := tc.osc.Verify(ctx, &ospb.VerifyRequest{})
-	if err != nil {
-		t.Fatalf("OS.Verify request failed: %s", err)
-	}
+	deadline := time.Now().Add(*timeout)
+	for time.Now().Before(deadline) {
+		r, err := tc.osc.Verify(ctx, &ospb.VerifyRequest{})
+		switch status.Code(err) {
+		case codes.OK:
+		case codes.Unavailable:
+			t.Log("Reboot in progress.")
+			time.Sleep(time.Minute)
+			continue
+		default:
+			t.Fatalf("OS.Verify request failed: %v", err)
+		}
+		// when noreboot is set to false, the device returns "in-progress" before initiating the reboot
+		if r.GetActivationFailMessage() == "in-progress" {
+			time.Sleep(time.Minute)
```

Comment:
maybe a log should be added here announcing the sleep like other logs in this func

Reply:
same as above, I've made this change and will fire off a test run, if successful, I'll raise a commit.

Reply:
commit has been raised